perceval/backend.py

```diff
@@ -510,6 +533,206 @@ def setup_cmd_parser(cls):
         raise NotImplementedError
 
 
+class BackendItemsGenerator:
+    """BackendItemsGenerator class.
+
+    This class provides a context manager that initializes a backend, used
+    to fetch items from data sources and/or archives in a transparent way. Items
+    are available in the class attributes `items`, while the summary of the
+    fetch executions can be accessed via the method `summary`.
+
+    :param backend_class: backend class to fetch items
+    :param backend_args: dict of arguments needed to fetch the items
+    :param category: category of the items to retrieve
+       If None, it will use the default backend category
+    :param manager: archive manager where the items will be retrieved
+    :param fetch_archive: If enabled, items are fetched from archives
+    :param filter_classified: remove classified fields from the resulting items. Note that
+        filter classified is not supported for archived items.
+    :param archived_after: return items archived after this date
+    """
+    def __init__(self, backend_class, backend_args, category, manager=None,
+                 fetch_archive=False, filter_classified=False, archived_after=None):
+        init_args = find_signature_parameters(backend_class.__init__,
+                                              backend_args)
+
+        if not fetch_archive:
+            archive = manager.create_archive() if manager else None
+            init_args['archive'] = archive
+            self.backend = backend_class(**init_args)
+            items = self.__fetch(backend_args, category, filter_classified, manager)
+        else:
+            self.backend = backend_class(**init_args)
+            items = self.__fetch_from_archive(manager, category, archived_after)
+
+        self.items = items
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def summary(self):
+        """Return the summary object in the backend"""
+        return self.backend.summary
+
+    def __fetch(self, backend_args, category, filter_classified=False,
+                manager=None):
+        """Fetch items using the given backend.
+
+        Generator to get items using the given backend class. When
```

Comment:
Part of this documentation should go to the description of the class too.

On this docstring itself, you shouldn't mention to the backend_class nor to the given backend. That was part of the old function doc.